### ISSUE TITLE
AKSv2 provisioning does not support the use of Kubenet with existing VNET

### ISSUE THREAD

**Issue by kkaempf** [rancher/dashboard#8750]
SURE-5140

**Setup**
- Rancher version: 2.7.2
- Browser type & version:

**Describe the bug**

The customer needs to provision AKS clusters in Rancher that are configured with Kubenet network option using an existing VNET subnet. This was supported in in earlier version of Rancher (using the AKS v1 provider) and is also officially supported in the Azure CLI.
https://docs.microsoft.com/en-us/azure/aks/configure-kubenet

Since Rancher 2.6.x this configuration can no longer be set: When selecting the "Kubenet" option in the AKS v2 provisioning UI there is no dropdown field to select an existing VNET. Instead, Rancher will configure a new VNET for each cluster.

This seems like a breaking change between AKS v1 and AKS v2 provisioning. It also blocks the customer who has a requirement of provisioning AKS clusters via the Rancher UI.

See https://github.com/rancher/rancher/issues/38560

**Comment by richard-cox** [comment]
Looks like we need both https://github.com/rancher/aks-operator/pull/163 and https://github.com/rancher/aks-operator/pull/158 merged before starting (see https://github.com/rancher/rancher/issues/38560#issuecomment-1493832862)

**Comment by mjura** [comment]
> Looks like we need both [rancher/aks-operator#163](https://github.com/rancher/aks-operator/pull/163) and [rancher/aks-operator#158](https://github.com/rancher/aks-operator/pull/158) merged before starting (see [rancher/rancher#38560 (comment)](https://github.com/rancher/rancher/issues/38560#issuecomment-1493832862))

Both PR https://github.com/rancher/aks-operator/pull/189 and https://github.com/rancher/aks-operator/pull/158 were merged

**Comment by gaktive** [comment]
Q3 is the plan per PM.

**Comment by kwwii** [comment]
We need clarification on how this is intended to work and on the differences between v1 and v2, in addition to how we fetch the list of vnet.

**Comment by gaktive** [comment]
@kkaempf is there anything written up to clarify what's different between the v1 & v2 versions?

Looks like the milestone was taken off the related backend issue so need to confirm milestone.

**Comment by kkaempf** [comment]
ping @mjura 👆🏻

**Comment by gaktive** [comment]
Based on concerns that we're diverging from the Azure setup, more planning would be required so this looks more like a Q4 issue now. 

@kkaempf and @mjura keep us in the loop based on activity here and then the related backend ticket(s).

**Comment by gaktive** [comment]
Related backend no longer has a milestone but SURE-5140 still points at Q3. Marked as blocked for now though again, this looks more like Q4.

**Comment by mjura** [comment]
We would like to change UI and `kubenet` in this same way as `azurecni` plus add new option which is called `OutboundType` with list `LoadBalancer` (default) and `UserDefined`
So for kubenet it is:
![kubenet](https://github.com/rancher/dashboard/assets/4455471/95141f71-c429-4f3e-9a8a-4a1a9992747e)
and it should looks like similar to Azure CNI
![azure-cni](https://github.com/rancher/dashboard/assets/4455471/25d46859-8bf7-4b22-8dca-a50e0b7f6515)

**Comment by mantis-toboggan-md** [comment]
Included in AKS ember->vue port https://github.com/rancher/dashboard/pull/9669

**Comment by cpinjani** [comment]
Blocked on https://github.com/rancher/dashboard/issues/10431

**Comment by valaparthvi** [comment]
https://github.com/rancher/rancher/issues/38560#issuecomment-2220440031

**Comment by mantis-toboggan-md** [comment]
@valaparthvi this looks like a duplicate of an issue @cpinjani reported recently - https://github.com/rancher/aks-operator/issues/574#issuecomment-2205326165 it was determined to be working as intended. The `virtualNetwork` field is null which the ui is correctly reflecting

**Comment by valaparthvi** [comment]
>@valaparthvi this looks like a duplicate of an issue @cpinjani reported recently - https://github.com/rancher/aks-operator/issues/574#issuecomment-2205326165 it was determined to be working as intended. The virtualNetwork field is null which the ui is correctly reflecting

In that case, I will close the issue. Thank you for clearing that up.
